src/connection/pool.rs

```diff
@@ -4,31 +4,42 @@
 //! R2D2 Connection Pool
 //!
 
-use crate::{Connection, ConnectionBuilder, FbError, Transaction};
-use rsfbclient_core::{FirebirdClient, FirebirdClientRemoteAttach};
+use super::{ConnectionConfiguration, FirebirdClientDbOps, FirebirdClientFactory};
+use crate::{Connection, FbError, Transaction};
 
-pub struct FirebirdConnectionManager<C: FirebirdClient> {
-    conn_builder: ConnectionBuilder<C>,
+/// A manager for connection pools. Requires the `pool` feature.
+pub struct FirebirdConnectionManager<F: FirebirdClientFactory> {
+    client_factory: F,
+    conn_conf: ConnectionConfiguration<<F::C as FirebirdClientDbOps>::AttachmentConfig>,
 }
 
-impl<C> FirebirdConnectionManager<C>
+type FactoryAssociatedConnectionConfig<F> = ConnectionConfiguration<
+    <<F as FirebirdClientFactory>::C as FirebirdClientDbOps>::AttachmentConfig,
+>;
+
+impl<F> FirebirdConnectionManager<F>
 where
-    C: FirebirdClient,
+    F: FirebirdClientFactory,
 {
-    pub fn new(conn_builder: ConnectionBuilder<C>) -> Self {
-        Self { conn_builder }
+    pub fn new(client_factory: F, conn_conf: FactoryAssociatedConnectionConfig<F>) -> Self {
+        Self {
+            client_factory,
+            conn_conf,
+        }
     }
 }
```

Comment:
I don't see a need to have the `conn_conf` as a field of the `FirebirdConnectionManager`, as it exists in all `FirebirdClientFactory` implementations.
I think it would be better to have a new function on the `FirebirdClientFactory` trait that returns a `&ConnectionConfiguration`, so it can be used in line 42 of this file.